Python/Modules/army_manager.py

```diff
@@ -96,3 +98,214 @@ def attack_with_unit(self, unit:Unit , position):
                 self.viking_manager.manage_unit(unit)
             case UnitTypeId.SIEGETANK | UnitTypeId.SIEGETANKSIEGED:
                 self.tank_manager.manage_unit(unit)
+
+
+    async def split_combat_units(self):
+        # Check whether we have units or not
+        if self.bot.units.empty:
+            return
+
+        all_enemies = self.bot.enemy_units
+
+        # Check if any enemies exist
+        if all_enemies.empty:
+            print("No enemies to split against")
+            return
+
+        # I'm not completely sure why, but sometimes checking whether enemies is empty or not is not enough
+        # I've put this check in to ensure that the first enemy has a position, as this ensures we can find an
+        # existing enemy with valid values
+        if all_enemies.first.position is None:
+            return
+
+        unit_nearest_cc = self.bot.units.closest_to(self.bot.start_location)
+
+        nearest_enemy = all_enemies.closest_to(unit_nearest_cc)
+
+        # Double check. Might be removable, but during testing it complained
+        if nearest_enemy is None:
+            return
+
+        # 10 is an arbitrary number. Could implement the recursive checking for closer units to the current closer units
+        enemy_squad = self.bot.enemy_units.closer_than(10, nearest_enemy)
+        enemy_tags = set(unit.tag for unit in enemy_squad)
+
+        alive_units = self.bot.units
+
+
+        # Checks whether enemy_squad is a subset of the cached enemy_squad
+        if self.is_squad_cached_and_valid(enemy_squad, alive_units):
+            print("already cached that enemy squad")
+            engagement = self.get_engagement_by_tags(enemy_tags)
```

Comment:
Could this be combined with is_squad_cached_and_valid?
I am thinking it could look something like:
`engagmement = self.get_engagement(enemy_squad, alive_units)
if engagement != none:
    print("already cached that enemy squad")
    for unit in engagment.counter units:
    ...
`